daemon/list.go

```diff
@@ -254,7 +254,7 @@ func (daemon *Daemon) foldFilter(ctx context.Context, view *container.View, conf
 	err := psFilters.WalkValues("exited", func(value string) error {
 		code, err := strconv.Atoi(value)
 		if err != nil {
-			return err
+			return errdefs.InvalidParameter(errors.Wrapf(err, "invalid filter 'exited=%s'", value))
```

Comment:
Looks like a potential opportunity to also have a `filters.GetInt(key, defaultValue)`

Reply:
I considered `GetInt` but decided not to implement it because at the time I thought it's only used for `stars` (didn't notice the `exited` 😄) which made it a bit awkward to consider how multiple integer values should be handled and generalize solution from one usage.

So, to reevaluate this one:
Currently we seem to have two usages for parsing filter as integer and both usages have different approach in how they handle multiple values.

1. `exited` - uses all values
2. `stars` - chooses the largest value (https://github.com/moby/moby/blob/e040a9dcf355540982ff95fab0aabbfdd91c9a2c/daemon/images/image_search.go#L49)

To handle both those cases, we would only be able to have something like `GetInts(key string, default []int) ([]int, error)` which would just make sure that all values for the given filter are valid integers and would return a parsed `[]int`, or a default if the filter is missing.

This would simplify the `exited` case, because it would reduce this code to
```go
filtExited, err := psFilter.GetInts("exited", nil)
if err != nil {
...
```

And for `stars` this would still require to loop through the result, but now the int parsing would be gone, and it would focus only on the `if iHasStar > hasStarFilter { hasStarFilter = iHasStar }` part.


It's still a win I think. WDYT?

Reply:
Perhaps that may work (definitely not for this PR though!). Will have to give it another (closer) look to see what everything would look like. The "multiple values" (and in many cases rather ambiguous) is rather awkward with the filters.